cmake/modules/FindRif.cmake

```diff
@@ -1,42 +1,23 @@
-
-if (APPLE)
-    set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS} -undefined dynamic_lookup")
-elseif (WIN32)
-    add_definitions(-DWIN32)
-endif()
-
-
-
-
-
-
-if (APPLE)
-    if (NOT RIF_LOCATION_LIB)
-        set (RIF_LOCATION_LIB ${RIF_LOCATION}/OSX/Bin/Release/x64)
-    endif ()
-    set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS} -undefined dynamic_lookup")
-
-    if (NOT RIF_LOCATION_INCLUDE)
-        set (RIF_LOCATION_INCLUDE ${RIF_LOCATION}/OSX/RadeonImageFilters)
-    endif ()
-elseif (WIN32)
-    if (NOT RIF_LOCATION_LIB)
-        set (RIF_LOCATION_LIB ${RIF_LOCATION}/Windows/Bin/Release/x64)
-	endif ()
-	
-	if (NOT RIF_LOCATION_INCLUDE)
-        set (RIF_LOCATION_INCLUDE ${RIF_LOCATION}/Windows/RadeonImageFilters)
-    endif ()
-	
-    add_definitions(-DWIN32)
-elseif (UNIX)
-    if (NOT RIF_LOCATION_LIB)
-        set (RIF_LOCATION_LIB ${RIF_LOCATION}/CentOS7/Bin/Release/x64)
-    endif ()
-
-    if (NOT RIF_LOCATION_INCLUDE)
-        set (RIF_LOCATION_INCLUDE ${RIF_LOCATION}/CentOS7/RadeonImageFilters)
-    endif ()
+macro(SET_RIF_VARIABLES dirName)
+    if(NOT RIF_LOCATION_LIB)
+        set(RIF_LOCATION_LIB ${RIF_LOCATION}/${dirName}/Bin/Release/x64)
+    endif()
+
+    if(NOT RIF_LOCATION_INCLUDE)
+        set(RIF_LOCATION_INCLUDE ${RIF_LOCATION}/${dirName}/RadeonImageFilters)
+    endif()
+endmacro(SET_RIF_VARIABLES)
+
+if(APPLE)
+    SET_RIF_VARIABLES(OSX)
+elseif(WIN32)
+    SET_RIF_VARIABLES(Windows)
+elseif(PLATFORM STREQUAL "ubuntu18.04")
+    SET_RIF_VARIABLES(Ubuntu18)
+elseif(PLATFORM STREQUAL "ubuntu16.04")
+    SET_RIF_VARIABLES(Ubuntu)
+elseif(PLATFORM STREQUAL "centos7")
```

Comment:
so this is ok.  Let's say I'm on centos 8 and maybe want to force it to use the centos7 build.  How would I do this?

Reply:
We can make PLATFORM variable as an option and let the user specify it otherwise calculate it automatically as it is now. In that case, we should consider renaming it

Reply:
Ok please do that.